ACS-3353: POST support for "other fields"

krdabrowski opened the pull request:
- adding method `isEqualTo()` allowing assertion of whole model with a different model.

krdabrowski committed:
ACS-3353: POST support for "other fields"
- adding method `isEqualTo()` allowing assertion of whole model with different model.

krdabrowski committed:
ACS-3353: POST support for "other fields"
- adding method `isEqualTo()` allowing assertion of whole model with different model.

tpage-alfresco reviewed (approved):
Looks really good.  It might be worth a warning on the method that the `equals` method needs to be implemented (since I assume most existing objects don't have it implemented), but maybe that's blindingly obvious.

krdabrowski committed:
ACS-3353: POST support for "other fields"
- adding method `isEqualTo()` allowing assertion of whole model with different model.

krdabrowski commented:
Thanks! I've added warning to javadoc.
